static/app/views/replays/detail/console/consoleMessage.tsx

```diff
@@ -0,0 +1,108 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+import {sprintf, vsprintf} from 'sprintf-js';
+
+import AnnotatedText from 'sentry/components/events/meta/annotatedText';
+import {getMeta} from 'sentry/components/events/meta/metaProxy';
+import {IconClose, IconWarning} from 'sentry/icons';
+import space from 'sentry/styles/space';
+import {BreadcrumbTypeDefault} from 'sentry/types/breadcrumbs';
+
+interface MessageFormatterProps {
+  breadcrumb: BreadcrumbTypeDefault;
+}
+
+/**
+ * Attempt to stringify
+ */
+function renderString(arg: string | number | boolean | Object) {
+  if (typeof arg !== 'object') {
+    return arg;
+  }
+
+  try {
+    return JSON.stringify(arg);
+  } catch {
+    return arg.toString();
+  }
+}
+
+/**
+ * Attempt to emulate the browser console as much as possible
+ */
+function MessageFormatter({breadcrumb}: MessageFormatterProps) {
+  // Browser's console formatter only works on the first arg
+  const [message, ...args] = breadcrumb.data?.arguments;
+
+  const isMessageString = typeof message === 'string';
+
+  // Assumption here is that Array types are == placeholders, which means that's the number of arguments we need
+  const placeholders = isMessageString
+    ? sprintf.parse(message).filter(parsed => Array.isArray(parsed))
+    : [];
+
+  // Retrieve message formatting args
+  const messageArgs = args.slice(0, placeholders.length);
+
+  // Attempt to stringify the rest of the args
+  const restArgs = args.slice(placeholders.length).map(renderString);
+
+  const formattedMessage = isMessageString
+    ? vsprintf(message, messageArgs)
+    : renderString(message);
+
+  // TODO(replays): Add better support for AnnotatedText (e.g. we use message
+  // args from breadcrumb.data.arguments and not breadcrumb.message directly)
+  return (
+    <Fragment>
+      <AnnotatedText
+        meta={getMeta(breadcrumb, 'message')}
+        value={[formattedMessage, ...restArgs].join(' ')}
+      />
+    </Fragment>
+  );
```

Comment:
bonus `<Fragment>`? i'm guessing at what eslint will say

```suggestion
  return (
    <AnnotatedText
      meta={getMeta(breadcrumb, 'message')}
      value={[formattedMessage, ...restArgs].join(' ')}
    />
  );
```